src/func.cc

```diff
@@ -946,6 +946,29 @@ void ProfileFunc(const vector<Value*>& args, Evaluator* ev, string*) {
   }
 }
 
+void VariableLocationFunc(const vector<Value*>& args, Evaluator* ev, string* s) {
+  string arg = args[0]->Eval(ev);
+  WordWriter ww(s);
+  for (StringPiece var : WordScanner(arg)) {
+    Symbol sym = Intern(var);
+    Var* v = ev->PeekVar(sym);
+    const Loc& loc = v->Location();
+    if (loc.filename) {
+      ww.Write(loc.filename);
+    } else {
+      ww.Write("<unknown>");
+    }
+    if (loc.lineno > 0) {
```

Comment:
Is `:0` the convention used elsewhere for unknown line numbers?

Reply:
I don't think there's a consistent pattern.  -1 would also work, but I don't think it matters but allowing it to be unsigned is easier here.